### DISCUSSION TITLE
Reply(forward) сообщения в ту же беседу

### DISCUSSION THREAD

**Discussion by wintermonth2298** [SevereCloud/vksdk#205] [Q&A] [answered]
Не получается ответить на сообщение в беседе (сделать форвард сообщения и добавить текст к этому форварду).
Пытаюсь сделать "ответ" из беседу в нее же саму, выбрав определенное сообщение пользователя.

Метод Forward у params.NewMessagesSendBuilder принимает интерфейс. По разному пробовал загонять данные, каждый раз получаю ошибку - incorrect json. Другого способа сделать Reply(forward) не обнаружил.

**Comment by SevereCloud** [comment, marked as answer]
Параметр forward в [messages.send](https://vk.com/dev.php?method=messages.send) принимает JSON-объект объект:

- `owner_id` — владелец сообщений. Стоит передавать, если вы хотите переслать сообщения из сообщества в диалог;
- `peer_id` — идентификатор места, из которого необходимо переслать сообщения;
- `conversation_message_ids` — массив `conversation_message_id` сообщений, которые необходимо переслать. В массив conversation_message_ids можно передать сообщения:
  - находящиеся в личном диалоге с ботом;
  - являющиеся исходящими сообщениями бота;
  - написанными после того, как бот вступил в беседу и появился доступ к сообщениям.
- `message_ids` — массив `id` сообщений;
- `is_reply` — ответ на сообщения. Стоит передавать, если вы хотите ответить на сообщения в том чате, в котором находятся сообщения. При этом в `conversation_message_ids`/`message_ids` должен находиться **только** один элемент.

Чтобы упростить себе жизнь существует объект `object.MessagesForward`, который можно передать в `Forward`

```go
b.Forward(object.MessagesForward{
        PeerID:                 123,
        ConversationMessageIDs: []int{456},
        IsReply:                true,
})
```
